app/src/main/kotlin/com/wire/android/ui/authentication/welcome/WelcomeViewModel.kt

```diff
@@ -43,15 +44,17 @@ class WelcomeViewModel @Inject constructor(
     }
 
     private fun checkNumberOfSessions() {
-        getSessions().let {
-            when (it) {
-                is GetAllSessionsResult.Success -> {
-                    isThereActiveSession =
-                        it.sessions.filter { it.session is AuthSession.Session.Valid }.isNullOrEmpty().not()
-                }
-                is GetAllSessionsResult.Failure.Generic -> {}
-                GetAllSessionsResult.Failure.NoSessionFound -> {
-                    isThereActiveSession = false
+        runBlocking {
```

Comment:
Idk what this `runBlocking` can do for the entire app but maybe we should us `withTimeout` to not block the thread for a long time?

Reply:
i removed the runBloaking and lunched the deep link handling in a coroutine